src/parser/demonhunter/havoc/modules/talents/Demonic.js

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import TalentStatisticBox from 'interface/others/TalentStatisticBox';
+import STATISTIC_ORDER from 'interface/others/STATISTIC_ORDER';
+import Events from 'parser/core/Events';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import SpellLink from 'common/SpellLink';
+
+/**
+ * Example Report: https://www.warcraftlogs.com/reports/23dHWCrT18qhaJbz/#fight=1&source=16
+ */
+
+const META_BUFF_DURATION_EYEBEAM = 10000;
+
+class Demonic extends Analyzer{
+
+  eyeBeamCasts = 0;
+  goodDeathSweep = 0;
+  eyeBeamTimeStamp = undefined;
+  counter = undefined;
+  badCasts = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTalent(SPELLS.DEMONIC_TALENT.id);
+    if (!this.active) {
+      return;
+    }
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.EYE_BEAM), this.getTimeStamp);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.DEATH_SWEEP), this.countCasts);
+  }
+
+  getTimeStamp(event) {
+    const hasMetaBuff = this.selectedCombatant.hasBuff(SPELLS.METAMORPHOSIS_HAVOC_BUFF.id, event.timestamp - 1000);
+
+    if (hasMetaBuff) {
+      return;
+    }
+    this.eyeBeamCasts += 1;
```

Comment:
this method does more than just get the timestamp, so it might make more sense to just name it `onEyeBeamCast`.